java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CommonRequestHandler.java

```diff
@@ -400,8 +401,10 @@ static Set<String> normalizeHeaders(Set<String> headers) {
 
     @SneakyThrows
     HttpRequestFactory getRequestFactory(HttpEventRequest request) {
-        // per connection request factory, abstracts auth ...
-        HttpTransport transport = new NetHttpTransport();
+
+        //q: right idea here? or google-http-client provides factory implementation, but whole point of DI fw is to avoid factories
```

Comment:
In theory yes, it will fit better if we have the Provider<HttpTransport> in the DI